src/main/java/WayofTime/bloodmagic/ritual/types/RitualMagnetic.java

```diff
@@ -68,8 +72,10 @@ public void performRitual(IMasterRitualStone masterRitualStone) {
                 while (i <= radius) {
                     while (k <= radius) {
                         BlockPos newPos = pos.add(i, j, k);
+                        Vec3d newPosVector = new Vec3d(newPos);
                         IBlockState state = world.getBlockState(newPos);
-                        ItemStack checkStack = state.getBlock().getPickBlock(state, null, world, newPos, null);
+                        RayTraceResult fakeRayTrace = world.rayTraceBlocks(MRSpos, newPosVector, false);
+                        ItemStack checkStack = state.getBlock().getPickBlock(state, fakeRayTrace, world, newPos, null);
```

Comment:
Rather than passing a null player, look at how the Ritual of Crushing handles a fake player

https://github.com/WayofTime/BloodMagic/blob/b441e7fc1e0ce7587d70ae140c83681933994371/src/main/java/WayofTime/bloodmagic/ritual/types/RitualCrushing.java#L290-L292